saleor/order/calculations.py

```diff
@@ -584,6 +586,23 @@ def order_line_unit_discount_type(
     return order_line.unit_discount_type
 
 
+def order_undiscounted_shipping(
+    order: Order,
+    manager: PluginsManager,
+    lines: Optional[Iterable[OrderLine]] = None,
+    force_update: bool = False,
+) -> TaxedMoney:
+    """Return the undiscounted shipping price of the order.
+
+    It takes into account all plugins.
+    If the prices are expired, call all order price calculation methods
+    and save them in the model directly.
+    """
+    currency = order.currency
+    order, _ = fetch_order_prices_if_expired(order, manager, lines, force_update)
```

Comment:
Other calculation functions in this module accept `database_connection_name`, which is passed to `fetch_order_prices_if_expired.` In read queries, this saves some calls to writer DB and uses replica. I see it's also missing in some discount-related functions above. In general, we should use it if possible.